examples/binary_examples/PyFstat_example_binary_mcmc_vs_grid_comparison.py

```diff
@@ -0,0 +1,296 @@
+        F2=signal_parameters["F2"],
+        Alpha=signal_parameters["Alpha"],
+        Delta=signal_parameters["Delta"],
+        period=point[0],
+        asini=point[1],
+        tp=point[2],
+        argp=point[3],
+        ecc=point[4],
+    )
+print(f"2Fstat computed on {grid_points.shape[0]} points")
+print("")
+print("Plotting results...")
+dim = len(search_keys)
+fig, ax = plt.subplots(dim, 1, figsize=(10, 10))
+for ind in range(dim):
+    a = ax.ravel()[ind]
+    a.grid()
+    a.set(xlabel=search_keys_label[ind], ylabel=r"$2 \mathcal{F}$", yscale="log")
+    a.plot(grid_points[:, ind], twoF_values, "o")
+    a.axvline(signal_parameters[search_keys[ind]], label="Injection", color="orange")
+plt.tight_layout()
+fig.savefig(os.path.join(outdir, "grid_twoF_per_dimension.pdf"))
+
+
+print("Performing MCMCSearch...")
+
+# Set up priors for the binary parameters
+theta_prior = {
+    "F0": signal_parameters["F0"],
+    "F1": signal_parameters["F1"],
+    "F2": signal_parameters["F2"],
+    "Alpha": signal_parameters["Alpha"],
+    "Delta": signal_parameters["Delta"],
+}
+
+for key in search_keys:
+    theta_prior.update(
+        {
+            key: {
+                "type": "unif",
+                "lower": 0.999 * signal_parameters[key],
+                "upper": 1.001 * signal_parameters[key],
+            }
+        }
+    )
+if circular_orbit:
+    for key in ["ecc", "argp"]:
+        theta_prior[key] = 0
+        search_keys.remove(key)
+# theta_prior["tp"]["lower"] = signal_parameters["tp"] - 0.5 * signal_parameters["period"]
+# theta_prior["tp"]["upper"] = (
+#    signal_parameters["tp"]
+#    + 0.5 * (1 + signal_parameters["argp"] / np.pi) * signal_parameters["period"]
+# )
+
+# ptemcee sampler settings - in a real application we might want higher values
+ntemps = 3
+log10beta_min = -1
+nwalkers = 100
+nsteps = [200, 200]  # [burnin,production]
+
+mcmcsearch = pyfstat.MCMCSearch(
+    label="mcmc_search",
+    outdir=outdir,
+    sftfilepattern=os.path.join(outdir, "*simulated_signal*sft"),
+    theta_prior=theta_prior,
+    tref=signal_parameters["tref"],
+    nsteps=nsteps,
+    nwalkers=nwalkers,
+    ntemps=ntemps,
+    log10beta_min=log10beta_min,
+    binary=True,
+)
+# walker plot is generated during main run of the search class
+mcmcsearch.run(
+    plot_walkers=True,
+    walker_plot_args={"plot_det_stat": True, "injection_parameters": signal_parameters},
+)
+mcmcsearch.print_summary()
+
+# call some built-in plotting methods
+# these can all highlight the injection parameters, too
+print("Making MCMCSearch {:s} corner plot...".format("-".join(search_keys)))
+mcmcsearch.plot_corner(truths=signal_parameters)
+print("Making MCMCSearch prior-posterior comparison plot...")
+mcmcsearch.plot_prior_posterior(injection_parameters=signal_parameters)
+print("")
+
+print("*" * 20)
+print("Quantitative comparisons:")
+print("*" * 20)
+
+# some informative command-line output comparing search results and injection
```

Comment:
Stopped reading here, but will run it once myself and if output is ok, will assume code behind it is ok too.